airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/check/DestinationCheckerWithoutGeneric.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.check
+
+import io.airbyte.cdk.load.command.DestinationCatalog
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.message.DestinationRecordStreamComplete
+import io.airbyte.cdk.load.message.InputRecord
+import io.airbyte.cdk.load.util.serializeToString
+import io.airbyte.cdk.load.write.WriteOperation
+import java.io.InputStream
+import java.io.PipedInputStream
+import java.io.PipedOutputStream
+import java.io.PrintWriter
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+/**
+ * A check operation that is run before the destination is used.
+ *
+ * * Implementors must provide a [check] method that validates the configuration.
+ * * Implementors may provide a [cleanup] method that is run after the check is complete.
+ * * [check] should throw an exception if the configuration is invalid.
+ * * [cleanup] should not throw exceptions.
+ * * Implementors should not perform any side effects in the constructor.
+ * * Implementors should not throw exceptions in the constructor.
+ * * Implementors should not inject configuration; only use the config passed in [check].
+ */
+interface DestinationCheckerWithoutGeneric {
+    fun check()
+    fun cleanup() {}
+}
+
+class DestinationCheckerSyncWithoutGeneric(
```

Comment:
I kept this because it was in the version with generic but I'm wondering if we would just delete this until we actually need it

Reply:
This would enable the destinations that use that default checker to migrate. I don't love the default checker because it basically requires implementing everything for sync (a heavier burden than check). Don't feel strongly either way.